packages/explat-client/src/internal/state.ts

```diff
@@ -3,20 +3,30 @@
  */
 import type { ExperimentAssignment } from '../types';
 import * as Validations from './validations';
-import * as ExperimentAssignments from './experiment-assignments';
 
-const experimentNameToExperimentAssignment: Record< string, ExperimentAssignment | undefined > = {};
+type Store = {
+	experimentNameToExperimentAssignment: Record< string, ExperimentAssignment | undefined >;
+};
```

Comment:
I went on a walk in the snow. I really don't see the benefit of this abstraction. Can you expand on this more or provide examples of how it is better and why?

Reply:
If you mean having an object around it, yeah not entirely necessary but does as a guide for adding further state - I like to keep everything together.

I'm happy to remove it.

Reply:
> as a guide for adding further state

Makes sense. As someone who will happily write a single case switch statement to "hint" to people that they should write their new case there... people won't do that, in my experience; they'll write an if-statement right above the switch :sigh:. It's probably better to wait until we have more state before wrapping it in an object.

Reply:
It'd all be pretty self-documenting with a simple class, but I'm done arguing on this PR. :slightly_smiling_face: